### PR TITLE
Iterable[pd.DataFrame] as output

### PR DESCRIPTION
**Pull request by kvnkho**
Very not sure about this implementation. Please review carefully.

### CODE REVIEW

File: fugue/_utils/interfaceless.py [hunk 1]
```diff
@@ -431,6 +433,20 @@ def count(self, df: pd.DataFrame) -> int:
         return df.shape[0]
 
 
+class _IterablePandasParam(_DataFrameParamBase):
+    def __init__(self, param: Optional[inspect.Parameter]):
+        super().__init__(param, "Iterable[pd.DataFrame]", "q")
+
+    def to_input_data(self, df: DataFrame) -> Iterable[pd.DataFrame]:
+        return iter(df.as_pandas())
```
Comment (goodwanghan) [inline fugue/_utils/interfaceless.py:441]: We should throw exception here, iterable of pd.dataframe should not be used as input.

Also the test didn't cover the input case, maybe we need to add a test to assert it throws

File: tests/fugue/utils/test_interfaceless.py [hunk 2]
```diff
@@ -104,12 +104,13 @@ def test_parse_function():
...
     _parse_function(f16, "^0e?(c|[dl]+)x*$", "n")
+    _parse_function(f33, "^$", "q")
 
 
 def test_function_wrapper():
-    for f in [f20, f21, f212, f22, f23, f24, f25, f26, f30, f31, f32]:
+    for f in [f20, f21, f212, f22, f23, f24, f25, f26, f30, f31, f32, f34]:
         df = ArrayDataFrame([[0]], "a:int")
-        w = FunctionWrapper(f, "^[ldsp][ldsp]$", "[ldsp]")
+        w = FunctionWrapper(f, "^[ldspq][ldspq]$", "[ldspq]")
```
Comment (goodwanghan) [inline tests/fugue/utils/test_interfaceless.py:113]: Places using FunctionWrapper need to update. You figured out how to update this test, but extension conversion logics need to be updated. And tests should be added

File: tests/fugue/utils/test_interfaceless.py [hunk 3]
```diff
@@ -324,3 +325,10 @@ def f32(
     e += list(a)
     arr = [[x["a"]] for x in e]
     return ArrayDataFrame(arr, "a:int").as_dict_iterable()
+
+def f33() -> Iterable[pd.DataFrame]:
+    pass
+
+def f34(e: pd.DataFrame, a: LocalDataFrame) -> Iterable[pd.DataFrame]:
+    a = ArrayDataFrame(a, "a:int").as_pandas()
+    return iter([e, a])
```
Comment (goodwanghan) [inline tests/fugue/utils/test_interfaceless.py:334]: You miss a new line here

**Comment by kvnkho** [comment]
Resolved changes. Still not sure everything is right.

Main question I have is if the `count` method is right or if it should raise an error similar to the `to_input_data` method in the interfaceless.py file.

**Comment by goodwanghan** [comment]
Fantastic

Just a minor thing:
Line 441 in interfaceless was not tested, you should remove 440 and 441, the default behavior is NotImplementedError, you don't need to write additional test for it.

Also you should add a t9 here 
https://github.com/fugue-project/fugue/blob/5c0e86a77b5f16ffb70e6d09ae9401128318f781/fugue_test/builtin_suite.py#L532
it's extremely simple

```python
def t9(df:pd.DataFrame) -> Iterable[pd.DataFrame]:
    incr()
    yield df
```

this is important because we need to make sure output_transform will realize the yield on `Iterable[pd.DataFrame]`

**Review by goodwanghan — APPROVED** [review]

**Comment by kvnkho** [comment]
I think this is finally done. Tests have been added to `test_convert_output_cotransform` and `test_convert_output_transform` to be consistent with Transformer and Cotransformer. Similarly, the regex has been changed to support Iterable[pd.DataFrame]. 

100% coverage. All unit tests passing.

**Pull request merged**